### PR TITLE
[C-1414] Implement mobile upload business logic

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Migrates upload sagas to audius-client/common\
- Consumes sagas in mobile
- Uses upload actions/selectors in mobile

Ends with a track getting uploaded!

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1414-implement-mobile-upload-logic

File: packages/mobile/src/screens/upload-screen/processTrackFile.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import type { UploadTrack } from './types'
+
+const ALLOWED_MAX_AUDIO_SIZE_BYTES = 250 * 1000 * 1000
+
+export const processTrackFile = (
+  trackFile: DocumentPickerResponse
+): UploadTrack => {
+  const { name, size } = trackFile
+  if (size && size > ALLOWED_MAX_AUDIO_SIZE_BYTES) {
+    throw new Error('File to large')
```
Comment (sliptype) [inline packages/mobile/src/screens/upload-screen/processTrackFile.ts:13]:
```suggestion
    throw new Error('File too large')
```

File: packages/mobile/src/screens/upload-screen/processTrackFile.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { newTrackMetadata } from '@audius/common'
+import type { DocumentPickerResponse } from 'react-native-document-picker'
+
+import type { UploadTrack } from './types'
+
+const ALLOWED_MAX_AUDIO_SIZE_BYTES = 250 * 1000 * 1000
```
Comment (sliptype) [inline packages/mobile/src/screens/upload-screen/processTrackFile.ts:6]: Is this a constraint on desktop too?

**Review by sliptype — APPROVED** [review]
Wow, amazing stuff! So everything from desktop basically just works on native? 

There is a type issue btw

File: packages/mobile/src/screens/upload-screen/processTrackFile.ts (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/upload-screen/processTrackFile.ts:6]: yes it is, going to move a lot of common utilities over, including extracting photo and other things from the selected image

**Comment by dylanjeffers** [comment]
> Wow, amazing stuff! So everything from desktop basically just works on native?
> 
> There is a type issue btw

Yeah astoundingly it all works!!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1414-implement-mobile-upload-logic

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1414-implement-mobile-upload-logic

**Pull request merged**